src/extension-runners/chromium.js

```diff
@@ -0,0 +1,310 @@
+/* @flow */
+
+/**
+ * This module provide an ExtensionRunner subclass that manage an extension executed
+ * in a Chromium-based browser instance.
+ */
+
+import path from 'path';
+import {promisify} from 'util';
+
+import {fs} from 'mz';
+import mkdirp from 'mkdirp';
+import {
+  LaunchedChrome,
+  launch as defaultChromiumLaunch,
+  default as ChromeLauncher,
+} from 'chrome-launcher';
+import WebSocket from 'ws';
+
+import {createLogger} from '../util/logger';
+import {TempDir} from '../util/temp-dir';
+import type {
+  ExtensionRunnerParams,
+  ExtensionRunnerReloadResult,
+} from './base';
+
+type ChromiumSpecificRunnerParams = {|
+   chromiumBinary?: string,
+   chromiumProfile?: string,
+   chromiumLaunch?: typeof defaultChromiumLaunch,
+|};
+
+export type ChromiumExtensionRunnerParams = {|
+  ...ExtensionRunnerParams,
+  // Chromium desktop CLI params.
+  ...ChromiumSpecificRunnerParams,
+|};
+
+const log = createLogger(__filename);
+
+const asyncMkdirp = promisify(mkdirp);
+
+/**
+ * Implements an IExtensionRunner which manages a Chromium instance.
+ */
+export class ChromiumExtensionRunner {
+  cleanupCallbacks: Set<Function>;
+  params: ChromiumExtensionRunnerParams;
+  chromiumInstance: LaunchedChrome;
+  chromiumLaunch: typeof defaultChromiumLaunch;
+  reloadManagerExtension: string;
+  wss: WebSocket.Server;
+  exiting: boolean;
+  _promiseSetupDone: ?Promise<void>;
+
+  constructor(params: ChromiumExtensionRunnerParams) {
+    const {
+      chromiumLaunch = defaultChromiumLaunch,
+    } = params;
+    this.params = params;
+    this.chromiumLaunch = chromiumLaunch;
+    this.cleanupCallbacks = new Set();
+  }
+
+  // Method exported from the IExtensionRunner interface.
+
+  /**
+   * Returns the runner name.
+   */
+  getName() {
+    return 'Chromium';
+  }
+
+  async run(): Promise<void> {
+    if (!this._promiseSetupDone) {
```

Comment:
Can `run` be called more than once ...? If not, I would throw if it is set. But that would affect line coverage (unless you add a dummy test that calls run twice...), so another alternative is to unconditionally assign to `_promiseSetupDone`.